scripts/utils/retry-func.sh

```diff
@@ -0,0 +1,28 @@
+retry() {
+  local status
+  local interval=1
+  local retry=0
+  local -r factor=2
+  local -r max_retries=10
+  while true; do
+    "$@" 2> /tmp/errors.txt && break
+    cat /tmp/errors.txt
+    status=$?
+    ((retry += 1))
+    if [ $retry -gt $max_retries ]; then
+      echo "error: Command failed and reached max retries" 1>&2
+      return $status
+    fi
+    echo "warning: Command failed and will retry" 1>&2
+
+    unauthorized_error=`cat /tmp/errors.txt |grep -ci "unauthorized"`
+    if [ $unauthorized_error -ne 0 ]; then
+      echo "error: Unauthorized error, wrong registry credentials provided, won't retry"
+      return 401
+    fi
```

Comment:
Let's simplify as:

```suggestion
    if grep -q -i "unauthorized" /tmp/errors.txt; then
      echo "error: Unauthorized error, wrong registry credentials provided, won't retry"
      return 401
    fi
```

Reply:
I'd rather use:
unauthorized_error=$(grep -ci "unauthorized" /tmp/errors.txt)
    if [ "$unauthorized_error" -ne 0 ]; then

which is way safer